### ISSUE TITLE
Error when importing an exported mask (v3.5.0.4)

### ISSUE THREAD

**Issue by alkhimey** [adventuregamestudio/ags#548]
I noticed there is a new feature that allows exporting masks. The issue here is that an exported mask can not be imported back as is.

![2018-12-21_21-52-11](https://user-images.githubusercontent.com/652603/50365128-d38f2200-057b-11e9-9a77-e0212a4dccb6.gif)

![image](https://user-images.githubusercontent.com/652603/50365132-d9850300-057b-11e9-8c43-4541618942c4.png)

**Comment by alkhimey** [comment]
Uploading the mask itself just in case this is not easily reproducible.

[test.zip](https://github.com/adventuregamestudio/ags/files/2704399/test.zip)

**Comment by morganwillcock** [comment]
I imagine this is the same issue that affects loading GIF images, that anything coming loaded through GDI will end up as a 32 bit ARGB Bitmap, regardless of what the format of the file is.

**Comment by ivan-mogilko** [comment]
>  that anything getting loaded through GDI will end up as a 32 bit ARGB Bitmap, regardless of what the format of the file is.

But if that was so, then would not the 8-bit mask import or 16-bit backgrounds would never work at all? 

I found what is wrong. I mistakenly assumed that if I tell it to create Bitmap of the indexed 8bit format it would also save it as one. But in reality it saves a PNG! You can tell this by viewing raw file contents.

**Comment by morganwillcock** [comment]
> But if that was so, then would not the 8-bit mask import or 16-bit backgrounds never work at all?

I guess that loading a BMP image bypasses GDI or some internal GDI decoder. You are right, an 8 bpp BMP file does end up as an 8 bpp Bitmap, whereas the 8bpp PNG ends up as a 32 bpp Bitmap.

> On a side note though, I keep wondering whether that is possible to teach Editor import 8-bit pngs as well?

I think this would require using another image loader or converting back to 8 bpp internally.

**Comment by ivan-mogilko** [comment]
I think this should work: cb7f005a7d3d800831ea87bf65d7cd8036e58e29.

**Comment by ivan-mogilko** [comment]
Okay, I found something confusing. Apparently calling Bitmap.Save *always* saves as PNG regardless of what file extension I type.

It saves indexed BMP if I do this:
<pre>
bmp.Save(fileName, ImageFormat.Bmp);
</pre>
With that I was able to reuse same bitmap as a room mask.

But the consequence of this is of course that it will save in BM format even if I choose non-bmp file extension.

So, if above code is a correct fix, then it will probably make sense to use more restrictive file filter in the save dialog (only bmps)?
